rest-server/engines/src/main/scala/com/actionml/engines/urnavhinting/URNavHintingAlgorithm.scala

```diff
@@ -64,8 +64,9 @@ class URNavHintingAlgorithm private (engine: URNavHintingEngine, initParams: Str
   with JsonParser {
 
   import URNavHintingAlgorithm._
+  import com.actionml.core.spark.SparkMongoSupport.syntax._
 
-  private var sparkContext: Validated[ValidateError, SparkContext] = _
+  private lazy val eventsDao = dataset.store.createDao[URNavHintingEvent](dataset.getIndicatorEventsCollectionName)
```

Comment:
Oh, no. The collections should be in the Dataset, not created here. There should be no change in any use of the DAOs. The Store will not longer read a collection into an RDD, the DAO will. But this should have no effect on what class creates the DAO. You are correct, the Algorithm only manages the model, which we have not formalized yet. It may be a DAO but is certainly not the input DAO like the Events DAO, which is the responsibility of a Dataset.